### PR TITLE
`anni`: Text Bricks CRUD

### LINKED ISSUE #338: `anni`: text bricks
Text bricks are predefined components that are used to create texts for annotations, namely

- a medication's patient-friendly drug class & indication and
- a medication-phenotype combination's implication & recommendation.

The creation of annotation texts is strictly limited to combinations of bricks to ensure consistency and enable easy multi-language support without the need of the maintainer having to know more than one supported language. Bricks can also include placeholders such as a given medication's name, but this won't be implemented with this issue.

### Tasks

- [x] database model for multi-language text bricks
- [x] CRUD-interface for text bricks
  - [x] list of bricks organized by category with settable display language
  - [x] bricks can be defined in multiple languages (list of available languages hardcoded)
- [x] add *text brick* and *annotation* to glossary
- [x] add in-app explanations & terminology

### PR DESCRIPTION
**Pull request by jannis-baum**
Closes #338

### CODE REVIEW

File: annotation-interface/components/BrickForm.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+} from '../common/constants';
+import {
+    ITextBrick,
+    ITextBrickTranslation,
+    translationIsValid,
+    translationsToArray,
+    translationsToMap,
+} from '../database/models/TextBrick';
+import SelectionPopover from './SelectionPopover';
+const BrickForm = ({
```
Comment (Benjamin-Frost) [inline annotation-interface/components/BrickForm.tsx:19]: You might be able to utilize a Form library here to make your life much easier. I can highly recommend [React Hook Form](https://react-hook-form.com/).

React Hook Form could definitely replace most of your state and also validators (`useEffect`s)

File: annotation-interface/pages/bricks/index.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            </FilterTabs>
+        </>
+    );
+};
+
+export default AllTextBricks;
+
+export const getServerSideProps = async () => {
+    await dbConnect();
+    const result = await TextBrick!.find({}).exec();
```
Comment (Benjamin-Frost) [inline annotation-interface/pages/bricks/index.tsx:98]: If you don't need hydration, you can speed up your queries quite a bit using [lean](https://mongoosejs.com/docs/tutorials/lean.html).

```suggestion
    const result = await TextBrick!.find().lean().exec();
```

**Review by Benjamin-Frost — APPROVED** [review]

File: annotation-interface/pages/bricks/index.tsx (hunk 2 above)
Comment (jannis-baum) [reply annotation-interface/pages/bricks/index.tsx]: Resolved in [6eb8835](https://github.com/hpi-dhc/PharMe/pull/336/commits/6eb88359a5990d54e911a9d6b8f87e2c5a8db005)

File: annotation-interface/pages/api/bricks/[id].ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                    new: true,
+                    runValidators: true,
+                });
+                if (!brick) {
+                    return res.status(400).json({ success: false });
+                }
+                res.status(200).json({ success: true, data: brick });
+            } catch (error) {
+                res.status(400).json({ success: false });
+            }
```
[the remark this answers is not in the document]
Comment (jannis-baum) [reply annotation-interface/pages/api/bricks/[id].ts]: [6eb8835](https://github.com/hpi-dhc/PharMe/pull/336/commits/6eb88359a5990d54e911a9d6b8f87e2c5a8db005)

File: annotation-interface/components/BrickForm.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    usage,
+    brick,
+}: {
+    usage: BrickUsage | null;
+    brick?: ITextBrick | null;
+}) => {
```
[the remark this answers is not in the document]
Comment (jannis-baum) [reply annotation-interface/components/BrickForm.tsx]: Resolved in [6eb8835](https://github.com/hpi-dhc/PharMe/pull/336/commits/6eb88359a5990d54e911a9d6b8f87e2c5a8db005)

File: annotation-interface/components/BrickForm.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    const deleteTranslation = (language: SupportedLanguage): void => {
+        setTranslations((prev) => {
+            const n = new Map(prev);
+            n.delete(language);
+            return n;
+        });
+    };
+    const save = async () => {
+        try {
+            const res = await fetch(`/api/bricks${id ? '/' + id : ''}`, {
```
[the remark this answers is not in the document]
Comment (jannis-baum) [reply annotation-interface/components/BrickForm.tsx]: Resolved in [bb217aa](https://github.com/hpi-dhc/PharMe/pull/336/commits/bb217aaa21e2ef6f95ac9b0ff044992052aad239)

**Pull request merged**